facebook-commerce-events-tracker.php

```diff
@@ -187,6 +193,128 @@ function( $item ) {
 			$this->pixel->inject_event( $event_name, $event_data, 'trackCustom' );
 		}
 
+
+		/**
+		 * Attempts to add a session variable to indicate that a product search event occurred.
+		 *
+		 * The session variable is used to catch search events that have a single result.
+		 * In those cases WooCommerce redirects customers to the product page instead of showing the search results.
+		 *
+		 * The plugin can't inject a Pixel event code on redirect responses, but it can check for the presence of the variable on the product page.
+		 *
+		 * This method is hooked to woocommerce_redirect_single_search_result which is triggered right before redirecting.
+		 *
+		 * @internal
+		 *
+		 * @since 2.0.6-dev.1
+		 *
+		 * @param bool $redirect whether to redirect to the product page
+		 * @return bool
+		 */
+		public function maybe_add_product_search_event_to_session( $redirect ) {
+
+			if ( $redirect && $this->search_event && $this->is_single_search_result() ) {
```

Comment:
I think here it needs checking if user logged in or not as WC session gets created anyways if user logged in or not
```php
if ( $redirect && is_user_logged_in() && $this->search_event && $this->is_single_search_result() ) {
```
That way the search event won't be reported if user not logged it which will pass the remaining QA Test

Reply:
@N-Molham thank you for reviewing!

If I understand correctly, you are saying that the event was reported when you tested while being logged out, is that right?

Do you remember if you had products in your cart when you ran the test? The session cookie is also added when you add a product to the cart.

In this case having the event sent for a logged out user is good. The ideal solution would be to send the event in all cases.

I used "(logged out)" in the test case description to highlight the behavior of the solution in scenario in which there is no customer session available and we can't pass the event data between requests. I should have made that intention explicit.

Right now, if the user is logged out but a session is available (because the customer added a product to the cart or other reason) we want to use that session and report the event.

Reply:
@wvega 

> If I understand correctly, you are saying that the event was reported when you tested while being logged out, is that right?

Yes, correct

> Do you remember if you had products in your cart when you ran the test? The session cookie is also added when you add a product to the cart.

With logged out state, I tried a different browser state (incognito mode on Chrome)

> Right now, if the user is logged out but a session is available (because the customer added a product to the cart or other reason) we want to use that session and report the event.

To be clear, when I tested the logged out state, I didn't add a product to cart before trying the search.

> I used "(logged out)" in the test case description to highlight the behavior of the solution in scenario in which there is no customer session available and we can't pass the event data between requests. I should have made that intention explicit.

I think either ways, it's good that it triggers the event even if user not logged in ... right?

Reply:
Yes, I think is good that it triggers the event even if the user is not logged in. 

I'm curious to find out why it was sent for you even though you were logged out and hadn't added a product to the cart. **We don't want** to cause WooCommerce to create a customer session if one doesn't already exists. Hosting providers usually disable cache when they see the session cookie, so we should be careful no to create it for this workaround.

I'll confirm that we are not creating a new session. Everything else seems ok based on my previous tests and your review results.

Reply:
I think that is different from hosting env to another, Maybe something in my server env triggers WooCommerce to create a session automatically upon the first visit ... let me try to debug it and see why and when a new session get created on my env